www/assets/js/webauthn.js

```diff
@@ -0,0 +1,111 @@
+// the following two functions are taken from https://stackoverflow.com/questions/16363419/how-to-get-binary-string-from-arraybuffer
+
+function BinaryToString(binary)
+{
+    var error;
+
+    try {
+        return decodeURIComponent(escape(binary));
+    } catch (_error) {
+        error = _error;
+        if (error instanceof URIError) {
+            return binary;
+        } else {
+            throw error;
+        }
+    }
+}
+
+function ArrayBufferToString(buffer)
+{
+    return BinaryToString(String.fromCharCode.apply(null, Array.prototype.slice.apply(new Uint8Array(buffer))));
+}
+
+function registrationButtonClick() {
+    navigator.credentials.create(publicKeyCredentialCreationOptions)
+    .then((cred) => {
+        console.log('NEW CREDENTIAL', cred);
+        document.getElementById('resp').value = cred.id;
+        var enc = new TextDecoder('utf-8');
+        document.getElementById('data').value = enc.decode(cred.response.clientDataJSON);
+        document.getElementById('attobj').value = btoa(ArrayBufferToString(cred.response.attestationObject));
+        document.getElementById('type').value = cred.response.type;
+        document.forms['regform'].submit();
+    })
+    .then((assertion) => {
+        console.log('ASSERTION', assertion);
+    })
+    .catch((err) => {
+        console.log('ERROR', err);
+    });
+}
+
+function authButtonClick() {
+    navigator.credentials.get(publicKeyCredentialRequestOptions)
+    .then((cred) => {
+        console.log('NEW CREDENTIAL', cred);
+        document.getElementById('resp').value = cred.id;
+        var enc = new TextDecoder('utf-8');
+        document.getElementById('data_raw_b64').value = btoa(ArrayBufferToString(cred.response.clientDataJSON));
+        document.getElementById('data').value = enc.decode(cred.response.clientDataJSON);
+        document.getElementById('authdata').value = btoa(ArrayBufferToString(cred.response.authenticatorData));
+    document.getElementById('sigdata').value = btoa(ArrayBufferToString(cred.response.signature));
```

Comment:
nitpick: indentation is off

Reply:
Indentation and spacing is incorrect in more places - I did not want to mix refactoring into this PR. I just moved the code from PHP and I would address format in another PR. Is that OK?